### PR TITLE
docs and licenses and castup, oh my

### PR DESCRIPTION
**Pull request by loriab**
## Description
docs and licenses and castup, oh my

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [x] few tweaks to the energy/grad docstring tables, #249 
- [x] improve molecule and fn return docs, #351 & #352 
- [x] update licenses in plugin template dir, since those are mixed py/c++ and awkward to do by script
- [x] hesitant replacement of cast-up with pyconstruct basis machinery. no differences detected in relevant output files (castup1/2/3, sapt2/4/5, pywrap-all), #285 
- [x] do `[s]` on seconds in sapt

## Status
- [x]  Ready to go

### CODE REVIEW

File: share/python/qcdb/molecule.py [hunk 1]
```diff
@@ -303,7 +303,9 @@ def save_string_xyz(self, save_ghosts=True):
             for i in range(self.natom()):
                 if self.Z(i):
                     N += 1
-        text = "%d\n" % (N)
+        text = ''
+        if save_natom:
+            text += "%d\n" % (N)
```
Comment (dgasmith) [inline share/python/qcdb/molecule.py]: Isn't the xyz file defined as """N\n\n x, y, x...""", should default to `True`?

File: src/lib/libscf_solver/hf.cc [hunk 2]
```diff
@@ -1529,9 +1529,11 @@ void HF::load_orbitals()
...
     boost::shared_ptr<BasisSet> dual_basis;
     if (basisname != options_.get_str("BASIS")) {
-        boost::shared_ptr<BasisSetParser> parser(new Gaussian94BasisSetParser(old_forced_puream));
-        molecule_->set_basis_all_atoms(basisname, "DUAL_BASIS_SCF");
-        dual_basis = BasisSet::construct(parser, molecule_, "DUAL_BASIS_SCF");
+        //boost::shared_ptr<BasisSetParser> parser(new Gaussian94BasisSetParser(old_forced_puream));
+        //molecule_->set_basis_all_atoms(basisname, "DUAL_BASIS_SCF");
+        //dual_basis = BasisSet::construct(parser, molecule_, "DUAL_BASIS_SCF");
+        dual_basis = BasisSet::pyconstruct_orbital(molecule_,
+        "BASIS", basisname, old_forced_puream);
```
Comment (dgasmith) [inline src/lib/libscf_solver/hf.cc]: Shouldn't this be "DUAL_BASIS_SCF"?

File: share/python/qcdb/molecule.py (hunk 1 above)
Comment (loriab) [inline share/python/qcdb/molecule.py]: So very true. And that was my first attempts to reconcile the psi4.Mol and qcdb.Mol functions. But the fact is that this fn is mostly used to generate a string to plop within `molecule mymol {...}`, and in that context the `N` is trouble. So better to leave psi4 alone and alter qcdb.

File: src/lib/libscf_solver/hf.cc (hunk 2 above)
Comment (loriab) [inline src/lib/libscf_solver/hf.cc]: I'm not actually sure. (Hence all the commented alarm in this section.) Testing the other way, too.

File: src/lib/libscf_solver/hf.cc [hunk 3]
```diff
@@ -1529,9 +1529,11 @@ void HF::load_orbitals()
...
     boost::shared_ptr<BasisSet> dual_basis;
     if (basisname != options_.get_str("BASIS")) {
-        boost::shared_ptr<BasisSetParser> parser(new Gaussian94BasisSetParser(old_forced_puream));
-        molecule_->set_basis_all_atoms(basisname, "DUAL_BASIS_SCF");
-        dual_basis = BasisSet::construct(parser, molecule_, "DUAL_BASIS_SCF");
+        //boost::shared_ptr<BasisSetParser> parser(new Gaussian94BasisSetParser(old_forced_puream));
+        //molecule_->set_basis_all_atoms(basisname, "DUAL_BASIS_SCF");
+        //dual_basis = BasisSet::construct(parser, molecule_, "DUAL_BASIS_SCF");
+        dual_basis = BasisSet::pyconstruct_orbital(molecule_,
+        "DUAL_BASIS_SCF", basisname, old_forced_puream);
```
Comment (loriab) [inline src/lib/libscf_solver/hf.cc]: Seems to work the same either way. So switched to the DUAL label. btw, you can do the following on two sep. files: ``git diff --color-words --no-index tests/sapt4/output.dat ../tests/sapt4/output.ref``

**Pull request merged**